src/app/features/psbt-signer/hooks/match-sats-with-outputs.ts

```diff
@@ -0,0 +1,84 @@
+import { isDefined } from '@shared/utils';
+
+import { PsbtInput } from './use-parsed-inputs';
+import { PsbtOutput } from './use-parsed-outputs';
+
+interface MatchedSatsWithOutput {
+  inputs: PsbtInput[];
+  output: PsbtOutput;
+}
+
+function consolidateSatsWithSameAddress(inputs: PsbtInput[]): PsbtInput[] {
+  const utxo: Record<string, PsbtInput> = {};
+
+  inputs.forEach(({ address, value, ...rest }) => {
+    if (isDefined(utxo[address])) {
+      utxo[address] = {
+        ...utxo[address],
+        value: utxo[address].value + value,
+      };
+    } else {
+      utxo[address] = { address, value, ...rest };
+    }
+  });
+  return Object.values(utxo);
+}
+
+export function matchSatsWithOuputs(
```

Comment:
maybe it would be better to add some test for this function? it would help to understand what this function does

Reply:
I do have a test included to test the matching ...in same name spec file.

Reply:
I can look at how to add more tests to it though to breakdown what is happening.